rllib/models/torch/model.py

```diff
@@ -0,0 +1,183 @@
+        save(self, path: str) -> None
+
+        # Load model weights from path
+        load(self, path: str) -> None
+
+    Examples:
+        >>> class MyCustomModel(TorchRecurrentModel):
+        ...     def __init__(self, input_size, output_size, recurrent_size):
+        ...         self.input_spec = SpecDict(
+        ...             {"obs": "batch, time, hidden"}, hidden=input_size
+        ...        )
+        ...         self.output_spec = SpecDict(
+        ...             {"logits": "batch, time, logits"}, logits=output_size
+        ...         )
+        ...         self.prev_state_spec = SpecDict(
+        ...             {"input_state": "batch, recur"}, recur=recurrent_size
+        ...         )
+        ...         self.next_state_spec = SpecDict(
+        ...             {"output_state": "batch, recur"}, recur=recurrent_size
+        ...         )
+        ...
+        ...         self.lstm = nn.LSTM(
+        ...             input_size, recurrent_size, batch_first=True
+        ...         )
+        ...         self.project = nn.Linear(recurrent_size, output_size)
+        ...
+        ...     def _unroll(self, inputs, prev_state, **kwargs):
+        ...         output, state = self.lstm(inputs["obs"], prev_state["input_state"])
+        ...         output = self.project(output)
+        ...         return TensorDict(
+        ...             {"logits": output}), TensorDict({"output_state": state}
+        ...         )
+
+    """
+
+    def __init__(self, config: ModelConfig) -> None:
+        RecurrentModel.__init__(self)
+        nn.Module.__init__(self)
+        TorchModelIO.__init__(self, config)
+
+    @override(RecurrentModel)
+    def _initial_state(self) -> TensorDict:
+        """Returns the initial recurrent state
+
+        This defaults to all zeros and can be overidden to return
+        nonzero tensors.
+
+        Returns:
+            A TensorDict that matches the initial_state_spec
+        """
+        return TensorDict(
+            tree.map_structure(
+                lambda spec: torch.zeros(spec.shape, dtype=spec.dtype),
+                self.initial_state_spec,
+            )
+        )
+
+
+class TorchModel(Model, nn.Module, TorchModelIO):
+    """The base class for non-recurrent pytorch models.
+
+    If implementing a custom pytorch model, you likely want to
+    inherit from this class.
+
+    Args:
+        config: The config used to construct the model
+
+    Required Attributes:
+        input_spec: SpecDict: Denotes the input keys and shapes passed to `_forward`
+        output_spec: SpecDict: Denotes the output keys and shapes returned from
+            `_forward`
+
+    Required Overrides:
+        # Define unrolling (forward pass) over a sequence of inputs
+        _forward(self, inputs: TensorDict, **kwargs)
+            -> TensorDict
+
+    Optional Overrides:
+        # Save model weights to path
+        save(self, path: str) -> None
+
+        # Load model weights from path
```

Comment:
_check_input and _check_output is also optional override

Reply:
fixed